gui/gui/src/TGWindow.cxx

```diff
@@ -117,14 +117,7 @@ TGWindow::~TGWindow()
 
 void TGWindow::SetWindowName(const char *name)
 {
-   if (!name && gDebug > 0) {
-      // set default frame names only when in debug mode
-      TString wname = ClassName();
-      wname += "::" + fName;
-      gVirtualX->SetWindowName(fId, (char *)wname.Data());
-   } else {
-      gVirtualX->SetWindowName(fId, (char *)name);
-   }
+   gVirtualX->SetWindowName(fId, (char *)name);
```

Comment:
I am confused.  If SetWindowName does not on windows, why keep it at (for windows)? I.e. why does "just" removing the alternate spelling when gDebug>0 helps?

Reply:
Let me rephrase this, @pcanal let me know if my wording still captures what you tried to express :-)

How can this fix anything, how was this failing? Was `ClassName()` crashing?

Reply:
The problem is that SetWindowName does work for top level windows but not internal ones on macOS (have not tested this on other platforms). Found this running:
gDebug = 1
.x hsimple.C++

Which then crashed in cocoa code trying to set a window name on an internal wid that does not accept this operation. The gDebug flag setting the Window name is wrong as in the routine we don't know if the window is top level or not. So instead of trying to find that out it's better not have gDebug change behavior beyond printing debug info.

Reply:
Thanks, Fons! Now, your change just calls `    gVirtualX->SetWindowName(fId, (char *)name); ` unconditionally. How can that prevent the crash you describe?

Reply:
OK I got it - it's a side effect: with your diff applied, a `name == nullptr` is still passed to `SetWindowName()`, while before, a non-null `name` was constructed. The first action in `TGCocoa::SetWindowName()` is to return early if `name == nullptr`, avoiding the `assert` at `CocoaPrivate.mm:107`: "GetDrawable, non-existing drawable requested" and, for debug builds, an error at like 97 and 103: "This drawable was deleted already", "This drawable not found among allocated/deleted drawables".

I.e. `TGWindow::fId` is wrong. It's `1`, coming from `gVirtualX->GetDefaultRootWindow()`, as part of `TGClient` calling `TGFrame()`. At that point, `CocoaPrivate::fDrawables` is empty, and so the window id 1 cannot be found.

I can actually reproduce the exact same issue with macOS 10.15. I'm fine with fixing this the way you suggest, but I'll add a comment to explain that this relies on `name == nullptr`.

Reply:
```suggestion
#ifdef R__MACOSX
   // MacOS fails to find the drawable of GetDefaultRootWindow(), only subsequent
   // windows ids can be found. See discussion here:
   // https://github.com/root-project/root/pull/6757#discussion_r518776154
   if (fId == gVirtualX->GetDefaultRootWindow())
      return;
#endif

   if (!name && gDebug > 0) {
      // set default frame names only when in debug mode
      TString wname = ClassName();
      wname += "::" + fName;
      gVirtualX->SetWindowName(fId, (char *)wname.Data());
   } else {
      gVirtualX->SetWindowName(fId, (char *)name);
   }
```